Lesson16/gozde/script.js

```diff
@@ -0,0 +1,263 @@
+
+function checkNumber(num) {
+    
+    if (num > 0) {
+        return "The number is positive.";
+    }
+    else if (num < 0) {
+        return "The number is negative.";
+    }
+    else {
+        return "The number is zero.";
+    }
+}
+
+// 6. Write a function that takes in a year and checks if it's a leap year (is divisible by 4 and not divisible by 100 OR divisible by 400) or not using if else statement.
+
+function isLeapYear(year) {
+    
+    if ((year % 4 === 0 && year % 100 !== 0) || year % 400 === 0) {
+        return true; 
+    } else {
+        return false; 
+    }
+}
+
+
+// 7. Write a function that takes in a temperature and checks if it's above or below freezing using if else statement.
+
+function checkFreezing(temperature) {
+    
+    if (temperature > 0) {
+        return "The temperature is above freezing.";
+    }
+    else if (temperature < 0) {
+        return "The temperature is below freezing.";
+    }
+    else {
+        return "The temperature is exactly freezing.";
+    }
+}
+
+// 8. Write a function that takes in a person's height and weight and checks if they are considered underweight, normal, overweight, or obese using if else statement.
+// Body Mass Index (BMI) is a person’s weight in kilograms (or pounds) divided by the square of height in meters (or feet). A high BMI can indicate high body fatness.
+//Formula: weight (kg) / [height (m)]2
+// Below 18.5	Underweight
+// 18.5 – 24.9	Healthy Weight
+// 25.0 – 29.9	Overweight
+// 30.0 and Above	Obesity
+
+function calculateBMI(height, weight) {
+    let heightInMeters = height / 100;
+
+    let bmi = weight / (heightInMeters * heightInMeters);
+
+    if (bmi < 18.5) {
+        return "Underweight";
+    } else if (bmi >= 18.5 && bmi <= 24.9) {
+        return "Healthy Weight";
+    } else if (bmi >= 25 && bmi <= 29.9) {
+        return "Overweight";
+    } else {
+        return "Obesity";
+    }
+}
+
+
+// 9. Write a function that takes in a person's age and checks if they are a child (age 0-12), teenager (age 13-19), adult (age 20-64), or senior citizen (age 65 and above) using if else statement.
+
+function checkAgeGroup(age) {
+    
+    if (age >= 0 && age <= 12) {
+        return "Child";
+    }
+    else if (age >= 13 && age <= 19) {
+        return "Teenager";
+    }
+    else if (age >= 20 && age <= 64) {
+        return "Adult";
+    }
+    else {
+        return "Senior Citizen";
+    }
+}
+
+
+// 10. Write a function that takes age as a number and whether they are a student (yes or no). If the age is less than 18 and the user is a student, print "You are a student and not old enough to vote". If the age is between 18 and 65 (inclusive) and the user is not a student, print "You are eligible to vote". If the age is greater than 65, print "You are a senior citizen". If the user is a student but their age is greater than or equal to 18, print "You are old enough to vote".
+
+function checkVotingEligibility(age, isStudent) {
```

Comment:
In programming, when you have a value that can be either yes or no (or similar), it's common to use a boolean instead. The solution using boolean would be like this:
```
function checkVotingEligibility(age, isStudent) {

    if (age < 18 && isStudent === "yes") {
        return "You are a student and not old enough to vote";
    }
    else if (age >= 18 && age <= 65 && !isStudent) { // !isStudent means if (isStudent === false)
        return "You are eligible to vote";
    }
    else if (age > 65) {
        return "You are a senior citizen";
    }
    else if (isStudent && age >= 18) { // In this case, condition means isStudent === true && age >= 18
        return "You are old enough to vote";
    }
    else {
        return "Invalid input";
    }
}

```